docs/src/main/asciidoc/spring-data-jpa-guide.adoc

```diff
@@ -0,0 +1,592 @@
+
+    List<Person> queryByJoinedIsAfter(Date date);
+
+    Collection<Person> readByActiveTrueOrderByAgeDesc();
+
+    Long countByActiveNot(boolean active);
+
+    List<Person> findTop3ByActive(boolean active, Sort sort);
+
+    Stream<Person> findPersonByNameAndSurnameAllIgnoreCase(String name, String surname);
+}
+----
+
+==== User defined queries
+
+User supplied queries contained in the `@Query` annotation. For example things like the following all work:
+
+[source,java]
+----
+public interface MovieRepository extends CrudRepository<Movie, Long> {
+
+    Movie findFirstByOrderByDurationDesc();
+
+    @Query("select m from Movie m where m.rating = ?1")
+    Iterator<Movie> findByRating(String rating);
+
+    @Query("from Movie where title = ?1")
+    Movie findByTitle(String title);
+
+    @Query("select m from Movie m where m.duration > :duration and m.rating = :rating")
+    List<Movie> withRatingAndDurationLargerThan(@Param("duration") int duration, @Param("rating") String rating);
+
+    @Query("from Movie where title like concat('%', ?1, '%')")
+    List<Object[]> someFieldsWithTitleLike(String title, Sort sort);
+
+    @Modifying
+    @Query("delete from Movie where rating = :rating")
+    void deleteByRating(@Param("rating") String rating);
+
+    @Modifying
+    @Query("delete from Movie where title like concat('%', ?1, '%')")
+    Long deleteByTitleLike(String title);
+
+    @Modifying
+    @Query("update Movie m set m.rating = :newName where m.rating = :oldName")
+    int changeRatingToNewName(@Param("newName") String newName, @Param("oldName") String oldName);
+
+    @Modifying
+    @Query("update Movie set rating = null where title =?1")
+    void setRatingToNullForTitle(String title);
+
+    @Query("from Movie order by length(title)")
+    Slice<Movie> orderByTitleLength(Pageable pageable);
+}
+----
+
+All methods that are annotated with `@Modifying` will automatically be annotated with `@Transactional`.
+
+==== More examples
+
+An extensive list of examples can be seen in the `integration-tests/spring-data` directory which is located inside the Quarkus source code.
+
+=== What is not currently supported
+
+* Methods of the `org.springframework.data.repository.query.QueryByExampleExecutor` Interface - if any of these are invoked, a Runtime exception will be thrown.
+* QueryDSL support. No attempt will be made to generate implementations of any of the QueryDSL related repositories.
+* Customizing the base repository for all repository interfaces in the code base.
+
+In Spring Data JPA this is done by registering a class that extends `org.springframework.data.jpa.repository.support.SimpleJpaRepository` however in Quarkus this class
+is not used at all (since all necessary plumbing is done at build time).
+We are exploring ways to provide similar functionality in the Quarkus support of Spring Data JPA.
+
+* `java.util.concurrent.Future` and related types are return types for methods.
```

Comment:
It's not clear what's not supported here.

Reply:
Is it better now?